### PR TITLE
Image in s3 Still working progress

### PR DESCRIPTION
**Pull request by dilippoudel**
I am working by creating a different component and importing to the profile-form page, it has distract the design, so, be patient, i will try to work when uploading to S3 works :)
Screen Shot 2021-01-12 at 20 33 38

### CODE REVIEW

File: src/components/JobseekerProfileForm/profileUpload.tsx [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+  })
+  const onDrop = (image: any) => {
+    setState({ images: image })
+  }
+  //uploads images to backend
+  const onClickHandler = () => {
+    const data = new FormData()
+    data.append('file', state.images[0])
+    axios.post('/s3/image', data, {})
+  }
```
Comment (Wyfy0107) [inline src/components/JobseekerProfileForm/profileUpload.tsx]: This is wrong. You don't **POST** the `data` to server. You should **POST** the `fileName` and `fileType` instead. After that you will get back a signed url from aws and then **POST** `data` to the url you get back

**Review by Wyfy0107 — CHANGES_REQUESTED** [review]

File: src/App.tsx [hunk 2]
```diff
@@ -6,15 +6,15 @@ import './scss/styles.scss'
...
-    axios.interceptors.request.use((config: any) => {
-      const token = LocalStorage.getToken()
-      if (token) {
-        config.headers['Authorization'] = `Bearer ${token}`
-      }
-      return config
-    })
-  }, [])
+  // useEffect(() => {
+  //   axios.interceptors.request.use((config: any) => {
```
Comment (Wyfy0107) [inline src/App.tsx]: you can take advantage of `config` object for the token removal. Here, `config` contains information about the request being sent, in which I believe there is a property `url` inside, and you can use this `url` to do conditioning

**Review by Wyfy0107 — CHANGES_REQUESTED** [review]

File: src/components/JobseekerProfileForm/profileUpload.tsx [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+import { AppState } from '../../redux/types'
+const ProfileUpload = () => {
+  const [state, setState]: any = React.useState({
+    images: [],
+    url: '',
+  })
+  // @ts-ignore
+  const userId = useSelector((state: AppState) => state.user.userInfo.id)
+  console.log('user is ', userId)
+  console.log(typeof userId)
```
Comment (Tsaikkari) [inline src/components/JobseekerProfileForm/profileUpload.tsx]: Remove

File: src/components/JobseekerProfileForm/profileUpload.tsx [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+
+  const onClickHandler = async () => {
+    console.log(state.images[0])
+    let file = state.images[0]
+    let fileParts = state.images[0].name.split('.')
+    let fileName = userId.toString()
+    let fileType = fileParts[1]
+    const data = new FormData()
+
+    console.log(state.images[0].name)
```
Comment (Tsaikkari) [inline src/components/JobseekerProfileForm/profileUpload.tsx]: Remove

File: src/components/JobseekerProfileForm/profileUpload.tsx [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+        let options = {
+          headers: {
+            'Content-Type': fileType,
+          },
+        }
+        axios
+          .put(signedRequest, file, options)
+          .then(result => {
+            console.log('result is ...', result)
+            console.log('response from s3')
```
Comment (Tsaikkari) [inline src/components/JobseekerProfileForm/profileUpload.tsx]: Remove

**Review by Tsaikkari — CHANGES_REQUESTED** [review]

**Review by khumair — APPROVED** [review]

**Review by khumair — APPROVED** [review]

**Review by khumair — APPROVED** [review]

**Review by khumair — DISMISSED** [review]

File: src/App.tsx [hunk 6]
```diff
@@ -9,9 +9,13 @@ const App = () => {
...
     axios.interceptors.request.use((config: any) => {
       const token = LocalStorage.getToken()
-      if (token) {
+      const regEx: any = /https:\/\/tindev-dev-test.s3.amazonaws.com\/tindev-image/g
+      let isMatch = config.url.match(regEx) ?? [] // exact same as config.url.match(regEx) ? config.url.match(regEx) : []
+
+      if (token && isMatch.length === 0) {
         config.headers['Authorization'] = `Bearer ${token}`
       }
+      console.log(config)
```
Comment (Tsaikkari) [inline src/App.tsx]: Remove

File: src/components/JobseekerProfileForm/profileUpload.tsx [hunk 7]
```diff
@@ -0,0 +1,79 @@
...
+import { Alert } from 'react-bootstrap'
+const ProfileUpload = () => {
+  const [state, setState]: any = React.useState({
+    images: [],
+    url: '',
+    message: '',
+  })
+  // @ts-ignore
+  const userId = useSelector((state: AppState) => state.user.userInfo.id)
+  console.log(typeof userId)
```
Comment (Tsaikkari) [inline src/components/JobseekerProfileForm/profileUpload.tsx]: Remove

**Review by Tsaikkari — APPROVED** [review]
:100:  Great work! Remove the console.logs though. :)

**Review by Tsaikkari — APPROVED** [review]

**Comment by dilippoudel** [comment]
who will merge this pr?

**Comment by Wyfy0107** [comment]
> who will merge this pr?

Take a look at the reviews and make changes accordingly.

**Comment by dilippoudel** [comment]
its already changes as review

File: src/components/JobseekerProfileForm/profileUpload.tsx [hunk 8]
```diff
@@ -0,0 +1,78 @@
...
+import { useSelector } from 'react-redux'
+import { AppState } from '../../redux/types'
+import { Alert } from 'react-bootstrap'
+const ProfileUpload = () => {
+  const [state, setState]: any = React.useState({
+    images: [],
+    url: '',
+    message: '',
+  })
+  // @ts-ignore
```
Comment (Wyfy0107) [inline src/components/JobseekerProfileForm/profileUpload.tsx]: Can you remove this ?

File: src/components/JobseekerProfileForm/profileUpload.tsx [hunk 9]
```diff
@@ -0,0 +1,78 @@
...
+  const userId = useSelector((state: AppState) => state.user.userInfo.id)
+  const onDrop = (image: any) => {
+    setState({ images: image })
+  }
+
+  const onClickHandler = async () => {
+    console.log(state.images[0])
```
Comment (Wyfy0107) [inline src/components/JobseekerProfileForm/profileUpload.tsx]: This also

**Review by Wyfy0107 — CHANGES_REQUESTED** [review]

File: src/index.tsx [hunk 10]
```diff
@@ -7,7 +7,7 @@ import axios from 'axios'
 import App from './App'
 import makeStore from './redux/store'
 
-axios.defaults.baseURL = 'https://tindev-dev-deploy.herokuapp.com'
+axios.defaults.baseURL = 'http://localhost:3000'
```
Comment (Wyfy0107) [inline src/index.tsx]: change this back !!

File: src/components/JobseekerProfileForm/profileUpload.tsx [hunk 11]
```diff
@@ -0,0 +1,76 @@
...
+  const onClickHandler = async () => {
+    let file = state.images[0]
+    let fileParts = state.images[0].name.split('.')
+    let fileName = userId.toString()
+    let fileType = fileParts[1]
+    const data = new FormData()
+    data.append('pic', state.images[0])
+
+    axios
+      .post('http://localhost:3000/s3/image', {
```
Comment (Wyfy0107) [inline src/components/JobseekerProfileForm/profileUpload.tsx]: '/s3/image'

**Review by Wyfy0107 — CHANGES_REQUESTED** [review]

**Comment by dilippoudel** [comment]
Done ✅

**Comment by dilippoudel** [comment]
Resolves Conflicts, now Ready to Merge :)

File: package.json [hunk 12]
```diff
@@ -17,18 +17,16 @@
     "axios": "^0.21.1",
     "bootstrap": "4.5.3",
     "node-sass": "4.14.1",
-    "react": "^16.3.1",
+    "react": "^17.0.1",
     "react-bootstrap": "1.4.0",
-    "react-dnd": "^5.0.0",
-    "react-dnd-html5-backend": "^3.0.2",
-    "react-dom": "^16.3.1",
```
Comment (Tsaikkari) [inline package.json]: We need these.

File: package.json [hunk 13]
```diff
@@ -17,18 +17,16 @@
...
+    "react-dom": "^17.0.1",
```
Comment (Tsaikkari) [inline package.json]: Remove this version

File: package.json [hunk 14]
```diff
@@ -17,18 +17,16 @@
...
+    "react-images-upload": "^1.2.8",
     "react-modern-calendar-datepicker": "^3.1.6",
     "react-redux": "^7.2.2",
     "react-router-bootstrap": "^0.25.0",
     "react-router-dom": "5.2.0",
     "react-scripts": "4.0.1",
     "react-spring": "^8.0.27",
-    "react-tag-input": "^6.5.2",
```
Comment (Tsaikkari) [inline package.json:31]: Put this back

File: package.json [hunk 15]
```diff
@@ -62,7 +60,6 @@
   },
   "devDependencies": {
     "@types/react-router-dom": "5.1.6",
-    "@types/react-tag-input": "^6.1.2",
```
Comment (Tsaikkari) [inline package.json]: This also

**Review by Tsaikkari — CHANGES_REQUESTED** [review]

File: src/redux/store.ts [hunk 16]
```diff
@@ -10,20 +10,15 @@ export const initState: AppState = {
     credential: { email: '', password: '' },
     userInfo: {
       role: '',
-      companyName: '',
-      companyInfo: '',
-      address: '',
-    },
-    info: {
-      jobPosts: [],
-      skills: [],
```
Comment (Tsaikkari) [inline src/redux/store.ts]: Put these back aswell.

**Review by Tsaikkari — CHANGES_REQUESTED** [review]

File: src/redux/store.ts (hunk 16 above)
Comment (dilippoudel) [reply src/redux/store.ts]: everything went wrong !!!

File: package.json (hunk 13 above)
Comment (dilippoudel) [reply package.json]: in the morning, you told to use v17, right

Comment (Tsaikkari) [reply package.json]: I told I needed to downgrade to 16. I think while resolving conflicts you copied your own package.json file from some branch and that's why my changes in the develop branch were overriden.

File: src/redux/store.ts (hunk 16 above)
Comment (Tsaikkari) [reply src/redux/store.ts]: Well, it's just 7 lines of code. ;) I hope lessons learned to day.

**Comment by dilippoudel** [comment]
@Tsaikkari Kirsi, could you check once again?

File: package.json [hunk 17]
```diff
@@ -17,18 +17,18 @@
     "axios": "^0.21.1",
     "bootstrap": "4.5.3",
     "node-sass": "4.14.1",
-    "react": "^16.3.1",
```
Comment (Tsaikkari) [inline package.json]: Put the v16 back and remove the v17. Or copy and paste the package.json file from the team's develop branch.

File: package.json [hunk 18]
```diff
@@ -17,18 +17,18 @@
...
     "react-dnd-html5-backend": "^3.0.2",
     "react-dom": "^16.3.1",
+    "react-images-upload": "^1.2.8",
     "react-modern-calendar-datepicker": "^3.1.6",
     "react-redux": "^7.2.2",
     "react-router-bootstrap": "^0.25.0",
     "react-router-dom": "5.2.0",
     "react-scripts": "4.0.1",
     "react-spring": "^8.0.27",
-    "react-tag-input": "^6.5.2",
```
Comment (Tsaikkari) [inline package.json:31]: This is still missing.

File: package.json [hunk 19]
```diff
@@ -62,7 +62,6 @@
   },
   "devDependencies": {
     "@types/react-router-dom": "5.1.6",
-    "@types/react-tag-input": "^6.1.2",
```
Comment (Tsaikkari) [inline package.json]: Just put it back...

**Review by Tsaikkari — CHANGES_REQUESTED** [review]

File: package.json (hunk 19 above)
Comment (dilippoudel) [reply package.json]: done

**Review by Tsaikkari — APPROVED** [review]

**Review by Wyfy0107 — APPROVED** [review]

**Pull request merged**